### PR TITLE
Update to the latest Vitess

### PR DESCRIPTION
**Pull request by dbussink**
On Vitess `main` there's a number of refactors that landed that greatly reduce the dependency surface area here and allow for things like cross compiling for the remaining dependencies to Windows.

### CODE REVIEW

File: handler.go [hunk 1]
```diff
@@ -17,12 +17,22 @@ import (
...
 	"vitess.io/vitess/go/sqltypes"
 	vitessquerypb "vitess.io/vitess/go/vt/proto/query"
+	"vitess.io/vitess/go/vt/sqlparser"
 	"vitess.io/vitess/go/vt/vterrors"
 )
 
 var errNotImplemented = errors.New("not implemented")
 
-func (s *Server) handler() *handler {
+const mysqlVersion = "8.0.34-psdbproxy"
```
Comment (dbussink) [inline handler.go:26]: This is what will be used both for the parser version (and how it deals with conditional comments) and for what we announce in the MySQL protocol bits.

File: server.go [hunk 2]
```diff
@@ -45,16 +45,12 @@ func (s *Server) Serve(l net.Listener) error {
...
 	})
 	if err != nil {
 		return err
 	}
 
-	if s.TLSConfig != nil {
-		listener.TLSConfig.Store(s.TLSConfig.Clone())
-		listener.RequireSecureTransport = true
-	}
-
```
Comment (dbussink) [inline server.go:57]: @mattrobenolt Was this used for anything? Not sure why it's cloned here and then set on the listener?

Comment (mattrobenolt) [reply server.go:57]: This was to support configuring the proxy for TLS. It wasn't hard yet inside pscale, but we might want it for some use case. If we remove it from here, we should drop it from the struct too.

**Review by mattrobenolt — APPROVED** [review]
I love it.

File: server.go (hunk 2 above)
Comment (dbussink) [reply server.go:57]: @mattrobenolt Dropped it from the struct. If we need anything, we can add something back later.

**Pull request merged**